src/V3ThreadPool.cpp

```diff
@@ -0,0 +1,85 @@
+// -*- mode: C++; c-file-style: "cc-mode" -*-
+//*************************************************************************
+// DESCRIPTION: Verilator: Thread pool for Verilator itself
+//
+// Code available from: https://verilator.org
+//
+//*************************************************************************
+//
+// Copyright 2005-2022 by Wilson Snyder.  This program is free software; you can
+// redistribute it and/or modify it under the terms of either the GNU
+// Lesser General Public License Version 3 or the Perl Artistic License
+// Version 2.0.
+//
+// Verilator is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+//*************************************************************************
+
+#include "config_build.h"
+
+#include "V3ThreadPool.h"
+
+#include "V3Ast.h"
+#include "V3Error.h"
+
+// The global thread pool
+V3ThreadPool v3ThreadPool;
+
+void V3ThreadPool::resize(unsigned n) {
```

Comment:
```suggestion
void V3ThreadPool::resize(unsigned n) VL_MT_SAFE {
```

Reply:
Note I'd like VL_MT_SAFE on both the forward declaration and actual declaration.